src/telemetry/mod.rs

```diff
@@ -33,8 +34,10 @@ pub fn new(
     metrics_retain_idle: Duration,
     taps: &Arc<Mutex<tap::Taps>>,
 ) -> (Sensors, tls_config_reload::Sensor, ServeMetrics) {
+    let process = process::Report::new(start_time);
```

Comment:
Was moving the construction of the `process::Report` here necessary for this change? I like the new factoring and I'm fine with landing it in this PR, but I just wanted to make sure that there isn't a semantic difference I'm missing...

Reply:
it's not strictly necessary, but now `metrics::new` takes several `Report` instances rather than constructing any reports internally.  This will fit well with subsequent changes.

Reply:
Okay, that's fine with me --- just wanted to make sure I hadn't overlooked something.